integration-tests/suites/runtime_config_file.go

```diff
@@ -0,0 +1,172 @@
+	connectionSuccess = s.Sensor().ExpectSameElementsConnections(s.T(), clientContainer, 10*time.Second, expectedConnections...)
+	s.Require().True(connectionSuccess)
+
+	s.deleteFile(runtimeConfigFile)
+	assert.AssertNoRuntimeConfig(s.T(), collectorIP)
+	// When the config file is deleted the config is reset and external IPs should continue to be disabled. There is no
+	// change in the expected connections.
+	common.Sleep(3 * time.Second) // Sleep so that collector has a chance to report connections
+	connectionSuccess = s.Sensor().ExpectSameElementsConnections(s.T(), clientContainer, 10*time.Second, expectedConnections...)
+	s.Require().True(connectionSuccess)
+
+	s.setExternalIpsEnable(runtimeConfigFile, true)
+	assert.AssertExternalIps(s.T(), true, collectorIP)
+	// The runtime config file is recreated with external IPs enabled. The unnormalized connection is reported as being active
+	// and the normalized connection is reported as being inactive.
+	expectedConnections = append(expectedConnections, activeUnnormalizedConnection, inactiveNormalizedConnection)
+	connectionSuccess = s.Sensor().ExpectSameElementsConnections(s.T(), clientContainer, 10*time.Second, expectedConnections...)
+	s.Require().True(connectionSuccess)
+
+	s.deleteFile(runtimeConfigFile)
+	assert.AssertNoRuntimeConfig(s.T(), collectorIP)
+	// The runtime config file is deleted again. This time there should be a change as external IPs transitions from being
+	// enabled to disabled. The normalized connection should be active and the unnormalized connection shoul be inactive.
+	expectedConnections = append(expectedConnections, activeNormalizedConnection, inactiveUnnormalizedConnection)
+	connectionSuccess = s.Sensor().ExpectSameElementsConnections(s.T(), clientContainer, 10*time.Second, expectedConnections...)
+	s.Require().True(connectionSuccess)
+
+	s.setExternalIpsEnable(runtimeConfigFile, false)
+	assert.AssertExternalIps(s.T(), false, collectorIP)
+	// The runtime config file is created one more time, this time with external IPs disabled. Since there is no change in the
+	// state, there should not be a change in the connections reported
+	common.Sleep(3 * time.Second) // Sleep so that collector has a chance to report connections
+	connectionSuccess = s.Sensor().ExpectSameElementsConnections(s.T(), clientContainer, 10*time.Second, expectedConnections...)
+	s.Require().True(connectionSuccess)
+
+	invalidConfig := "asdf"
+	s.setRuntimeConfig(runtimeConfigFile, invalidConfig)
+	assert.AssertExternalIps(s.T(), false, collectorIP)
+	// Testing an invalid configuration. There should not be a change in the configuration or reported connections
+	common.Sleep(3 * time.Second) // Sleep so that collector has a chance to report connections
+	connectionSuccess = s.Sensor().ExpectSameElementsConnections(s.T(), clientContainer, 10*time.Second, expectedConnections...)
+	s.Require().True(connectionSuccess)
```

Comment:
Couldn't this be split into multiple test cases? Reviewing this test is borderline impossible.

Reply:
It could be, but I think this is better. This tests for case when external IPs is enabled after the config file is deleted as well as the case when external IPs is disabled after the config file is deleted. It tests for deleting the config file when external IPs is enabled and it tests for deleting the config file with external IPs is disabled. These cases could be split into different tests, but it would take longer to run. Would better comments make it easier to understand what is going on in the test?

Reply:
> Couldn't this be split into multiple test cases? Reviewing this test is borderline impossible.

It's not a problem if they take a couple more seconds, the problem is that you have 8, highly condensed, nearly identical blocks of code. This makes it really hard to understand what is actually changing from one block to the next, because I can't see what the difference is. This, in turn, makes it really hard for me to give meaningful feedback on the test, because I can't understand it.

The intent of the test should be obvious from a glance at the code.

> Would better comments make it easier to understand what is going on in the test?

If anything, the existing comments are making the code even denser and harder to understand. Try doing shorter comments at the start of each block, don't go into deep details of what the block is doing, just give a hint of what it'll do and try to read the block of code from the perspective of a person that is going through it for the first time, ask yourself:
- Is it easy to see what has changed in this part of the test?
- Can I easily spot the intent of the block?
- Is it distinct enough from a previous, similar block?

Reply:
I have split the test into three tests.